WordPress/wordpress-develop
# Font Directory uploads ignore `subdir` property in post meta.

Records the sub-directory font files are stored in in the post meta.

Currently uploading a font to `fonts/sub-dir/filename.woff` only records `filename.woff` in the post meta (while the file placement is in the specified directory. This prevents the deletion of hte font files when the wp_font_face post object is deleted.

This modifies the post meta storage to include the sub-directory to ensure the files are deleted along with posts.

A mini plugin to demonstrate the issue:

```php
add_filter( 'font_dir', function ( $font_dir ) {
        /*
         * Change sub-directory every 5 minutes.
         *
         * POC only, a more likely use is to get the `fontFamily` from the API
         * request or the currently active theme and use either of those values
         * to generate a unique sub-directory.
         *
         * The POC is simply to allow for eaiser testing by setting a new
         * sub-directory every few minutes.
         */
        $subdir = '/' . (string) floor( time() / 30 );
        $font_dir['subdir'] = $subdir;
        $font_dir['path'] .= $subdir;
        $font_dir['url'] .= $subdir;
        return $font_dir;
} );
```

**reproduction steps (trunk)**

1. Add mini-plugin to mu-plugins
2. Upload/install a font from Google
3. Wait 30 seconds or so (for a tick of the clock in the plugin above to change the directory)
4. Delete the font from the font-library
5. Observe the file remains on the file system

**Testing PR**

1. Add mini-plugin to mu-plugins
2. Upload/install a font from Google
3. Wait 30 seconds or so (for a tick of the clock in the plugin above to change the directory)
4. Delete the font from the font-library
5. Observe the file is removed from the file system


https://core.trac.wordpress.org/ticket/61297
